vcs/src/main/java/org/openjdk/skara/vcs/git/GitRepository.java

```diff
@@ -919,16 +919,28 @@ public void prune(Branch branch, String remote) throws IOException {
     }
 
     @Override
-    public Hash mergeBase(Hash first, Hash second) throws IOException {
+    public Optional<Hash> mergeBaseOptional(Hash first, Hash second) throws IOException {
         try (var p = capture("git", "merge-base", first.hex(), second.hex())) {
-            var res = await(p);
```

Comment:
await(p) basically does p.await() and then throws IOException if res.status() is non zero.